catalog/spatial/geocoding/spatial-geocoding-lucene/src/test/java/org/codice/ddf/spatial/geocoding/query/GazetteerQueryCatalogTest.java

```diff
@@ -0,0 +1,324 @@
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.codice.ddf.spatial.geocoding.GeoCodingConstants;
+import org.codice.ddf.spatial.geocoding.GeoEntry;
+import org.codice.ddf.spatial.geocoding.GeoEntryMetacardType;
+import org.codice.ddf.spatial.geocoding.GeoEntryQueryException;
+import org.codice.ddf.spatial.geocoding.context.NearbyLocation;
+import org.junit.Before;
+import org.junit.Test;
+
+public class GazetteerQueryCatalogTest {
+
+  private static final int RADIUS_IN_KM = 100;
+
+  private static final int MAX_RESULTS = 10;
+
+  private static final String BOSTON_WKT = "POINT (-71.0595703125 42.35771940022451)";
+
+  private static final String NEAR_BOSTON_WKT = "POINT(-71.07124328613281 42.353671973455704)";
+
+  private static final String BAD_WKT = "badWkt";
+
+  private static final String QUERY_STRING = "queryString";
+
+  private static final String BOSTON = "Boston";
+
+  private static final String USA_COUNTRY_CODE = "USA";
+
+  private static final MetacardType GEO_ENTRY_METACARD_TYPE =
+      new MetacardTypeImpl(
+          "GeoEntryMetacardType",
+          Arrays.asList(new LocationAttributes(), new GeoEntryMetacardType()));
+
+  private static final FilterBuilder FILTER_BUILDER = new GeotoolsFilterBuilder();
+
+  private GazetteerQueryCatalog queryCatalog;
+
+  private CatalogFramework catalogFramework;
+
+  @Before
+  public void setUp() throws Exception {
+    catalogFramework = mock(CatalogFramework.class);
+    QueryResponse queryResponse = generateQueryResponseFromMetacard(generateGeoNamesMetacard());
+    when(catalogFramework.query(any(QueryRequest.class))).thenReturn(queryResponse);
+    queryCatalog = new GazetteerQueryCatalog(catalogFramework, FILTER_BUILDER);
+  }
+
+  @Test
+  public void testQuery() throws Exception {
+    List<GeoEntry> geoEntryList = queryCatalog.query(QUERY_STRING, 1);
+    assertThat(geoEntryList.size(), is(1));
+    GeoEntry geoEntry = geoEntryList.get(0);
+    assertThat(geoEntry.getCountryCode(), is(USA_COUNTRY_CODE));
+    assertThat(geoEntry.getName(), is(BOSTON));
+    assertThat(geoEntry.getFeatureCode(), is("PPL"));
+    assertThat(geoEntry.getLatitude(), is(42.35771940022451));
+    assertThat(geoEntry.getLongitude(), is(-71.0595703125));
+    assertThat(geoEntry.getPopulation(), is(123456789L));
+  }
+
+  @Test
+  public void testQueryEmptyMetacard() throws Exception {
+    QueryResponse queryResponse = generateQueryResponseFromMetacard(generateEmptyMetacard());
+    when(catalogFramework.query(any(QueryRequest.class))).thenReturn(queryResponse);
+    List<GeoEntry> geoEntryList = queryCatalog.query(QUERY_STRING, 1);
+    GeoEntry geoEntry = geoEntryList.get(0);
+    assertThat(geoEntry.getName(), nullValue());
+  }
+
+  /* @Test
```

Comment:
Should these be removed? Do we need a JIRA for working on these at some future time?

Reply:
uncommented these